src/new/swapConnection.ts

```diff
@@ -0,0 +1,95 @@
+/**
```

Comment:
I don't think the folder `new` is the right folder for it. I would create a new folder for `swapService`, or something like that, to keep the swap service files.